met-web/src/components/permissionsGate/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { cloneElement } from 'react';
+import { useAppSelector } from 'hooks';
+
+interface HasPermissionProps {
+    permissions: string[];
+    scopes: string[];
+}
+const hasPermission = ({ permissions, scopes }: HasPermissionProps) => {
+    const scopesMap: { [scope: string]: boolean } = {};
+
+    scopes.forEach((scope) => {
+        scopesMap[scope] = true;
+    });
+
+    return permissions.some((permission) => scopesMap[permission]);
+};
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+interface PermissionsGateProps {
+    children: React.ReactElement<any, any>;
+    RenderError?: () => React.ReactElement<any, any>;
+    errorProps?: any;
+    scopes: string[];
+}
+export function PermissionsGate({
+    children,
+    RenderError,
+    errorProps,
+    scopes = [],
```

Comment:
Would be great if u can write some comments about it..
the code is totally neat , good job..
still some detailed comment will be helpful

Reply:
Will handle in next PR. Thank you